packages/dev-container/src/electron-browser/container-connection-contribution.ts

```diff
@@ -59,46 +64,85 @@ export class ContainerConnectionContribution extends AbstractRemoteRegistryContr
         });
     }
 
+    canHandle(uri: URI): MaybePromise<boolean> {
+        return uri.scheme === DEV_CONTAINER_WORKSPACE_SCHEME;
+    }
+
+    async openWorkspace(uri: URI, options?: WorkspaceInput | undefined): Promise<void> {
+        const filePath = new URLSearchParams(uri.query).get(DEV_CONTAINER_PATH_QUERY);
+
+        if (!filePath) {
+            throw new Error('No devcontainer file specified for workspace');
+        }
+
+        const devcontainerFiles = await this.connectionProvider.getDevContainerFiles(uri.path.toString());
+        const devcontainerFile = devcontainerFiles.find(file => file.path === filePath);
+
+        if (!devcontainerFile) {
+            throw new Error(`Devcontainer file at ${filePath} not found in workspace`);
+        }
+
+        return this.doOpenInContainer(devcontainerFile, uri.toString());
+    }
+
+    async getWorkspaceLabel(uri: URI): Promise<string | undefined> {
+        const containerFilePath = new URLSearchParams(uri.query).get(DEV_CONTAINER_PATH_QUERY);
+        if (!containerFilePath) {
+            return;
+        };
+        const files = await this.connectionProvider.getDevContainerFiles(uri.path.toString());
+        const devcontainerFile = files.find(file => file.path === containerFilePath);
+        return `${uri.path.base} @ ${devcontainerFile?.name} [Dev Container]`;
```

Comment:
I personally find this way of representing the information a bit to clutered. I would prefer something like in VSCode (e.g. `{uri.path.base} [Dev Container: ${devcontainerFile?.name}]`.
But this is of course only my personal feeling. WDYT?

Also would it be possible to render the output of the `getWorkspaceLabel()` method in the title bar of the application? Currently this does not display the indicator that it is a dev container in the title for me.

Reply:
This is exactly how its displayed for me in vscode. So i just adapted that. But im totally open to any other formats

Regarding your second point thats something Jan is doing in this [PR](https://github.com/eclipse-theia/theia/pull/14571#event-15506041995)